django/core/mail/message.py

```diff
@@ -100,22 +100,62 @@ def forbid_multi_line_headers(name, val, encoding):
     return str(name), val
 
 
+def split_addr(addr, encoding):
+    """
+    Splits an address into local part and domain, properly encoded.
+    """
+    if '@' in addr:
+        localpart, domain = addr.split('@', 1)
+        localpart = Header(localpart, encoding).encode()
```

Comment:
This always encodes the local part with MIME word encoding, so `to@example.com` becomes something like `=?utf-8?q?to?=@example.com`. If we want to avoid that, we could do something like:
```
try:
    localpart.encode('ascii')
except UnicodeEncodeError:
    localpart = Header(localpart, encoding).encode()
```

this would make sure that simple localparts are kept simple and can be easily unit tested against by end users who test e-mail headers (`message.get('from')` etc.). That would also render a few unit test changes from Django unnecessary, making this PR a bit simpler. Any thoughts on this, anyone?